CMakeLists.txt

```diff
@@ -1,227 +1,182 @@
-	set(ZENOHC_INSTALL_STATIC_LIBRARY TRUE CACHE BOOL "Install static library")
-endif()
-message(STATUS "CMAKE_INSTALL_PREFIX: ${CMAKE_INSTALL_PREFIX}")
-message(STATUS "Configurations allowed to install: ${ZENOHC_INSTALL_CONFIGURATIONS}")
-message(STATUS "Static library will be installed: ${ZENOHC_INSTALL_STATIC_LIBRARY}")
-include(GNUInstallDirs)
-set(CMAKE_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/zenohc")
-install(FILES ${CMAKE_BINARY_DIR}/${libzenohc} CONFIGURATIONS ${ZENOHC_INSTALL_CONFIGURATIONS} DESTINATION ${CMAKE_INSTALL_LIBDIR})
-if (${ZENOHC_INSTALL_STATIC_LIBRARY})
-	install(FILES ${CMAKE_BINARY_DIR}/${libzenohc_static} CONFIGURATIONS ${ZENOHC_INSTALL_CONFIGURATIONS} DESTINATION ${CMAKE_INSTALL_LIBDIR})
+# Workaroud for https://github.com/rust-lang/cargo/issues/5045
+# mentioned in https://github.com/eclipse-zenoh/zenoh-c/issues/138
+# If it's fixed, do not forget to correct PackageConfig.cmake.in also
+set_target_properties(zenohc PROPERTIES IMPORTED_NO_SONAME TRUE)
+
+function(set_target_imported_locations target libname)
+	set_target_properties(${target}
+		PROPERTIES 
+		IMPORTED_GLOBAL TRUE
+		IMPORTED_LOCATION ${cargo_binary_dir}/${libname}
+		IMPORTED_LOCATION_DEBUG ${cargo_binary_dir_debug}/${libname}
+		IMPORTED_LOCATION_RELEASE ${cargo_binary_dir_release}/${libname}
+		IMPORTED_LOCATION_MINSIZEREL ${cargo_binary_dir_release}/${libname}
+		IMPORTED_LOCATION_RELWITHDEBINFO ${cargo_binary_dir_release}/${libname}
+	)
+endfunction()
+
+function(set_target_imported_implib target libname)
+	set_target_properties(${target}
+		PROPERTIES 
+		IMPORTED_GLOBAL TRUE
+		IMPORTED_IMPLIB ${cargo_binary_dir}/${libname}
+		IMPORTED_IMPLIB_DEBUG ${cargo_binary_dir_debug}/${libname}
+		IMPORTED_IMPLIB_RELEASE ${cargo_binary_dir_release}/${libname}
+		IMPORTED_IMPLIB_MINSIZEREL ${cargo_binary_dir_release}/${libname}
+		IMPORTED_IMPLIB_RELWITHDEBINFO ${cargo_binary_dir_release}/${libname}
+	)
+endfunction()
+
+set_target_imported_locations(zenohc_static ${staticlib})
+set_target_imported_locations(zenohc ${dylib})
+if(DEFINED implib)
+	set_target_imported_implib(zenohc ${implib})
 endif()
-install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/" CONFIGURATIONS ${ZENOHC_INSTALL_CONFIGURATIONS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/zenohc.pc.in" "${CMAKE_CURRENT_SOURCE_DIR}/zenohc.pc" @ONLY)
 
-if(APPLE OR UNIX)
-	install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/zenohc.pc" CONFIGURATIONS ${ZENOHC_INSTALL_CONFIGURATIONS} DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig" OPTIONAL)
+# Define include directories for library targets
+status_print(source_include_dir)
+status_print(cargo_generated_include_dir)
+target_include_directories(zenohc_static INTERFACE ${source_include_dir})
+target_include_directories(zenohc INTERFACE ${source_include_dir})
+if(DEFINED cargo_generated_include_dir)
+	file(MAKE_DIRECTORY ${cargo_generated_include_dir})
```

Comment:
Maybe this line is not necessary. CMake itself allows having nonexistent include directory which will be created at any time (build, post-build)....

Reply:
No, this line is required. CMake can create directories on build stage, but the trick is that this directory is required on confifguration stage:
```
CMake Error in tests/CMakeLists.txt:
  Imported target "zenohc" includes non-existent path

    "/home/milyin/ZS/foo/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
```

Reply:
Ahha, this necessary is for INTERFACE libraries.... I'll remember this